packages/api/src/mappings/apple/sleep.ts

```diff
@@ -19,36 +19,54 @@ export function mapDataToSleep(data: AppleHealth, hourly: boolean) {
       sleepPayload.durations = {
         in_bed_seconds: appleItem.value,
       };
-    }
 
-    if (appleItem.type === SleepType.awake) {
-      sleepPayload.durations = {
-        awake_seconds: appleItem.value,
-      };
+      sleep.push(sleepPayload);
     }
+  };
 
-    if (appleItem.type === SleepType.core) {
-      sleepPayload.durations = {
-        light_seconds: appleItem.value,
-      };
-    }
+  data.HKCategoryValueSleepAnalysis?.forEach(appleSleepItem => addInBedToSleep(appleSleepItem));
+  data.HKCategoryValueSleepAnalysis?.forEach(appleSleepItem => {
+    if (appleSleepItem.type !== SleepType.inBed) {
+      const index = sleep.findIndex(
+        item => item.start_time && item.start_time <= appleSleepItem.date
+      );
 
-    if (appleItem.type === SleepType.deep) {
-      sleepPayload.durations = {
-        deep_seconds: appleItem.value,
-      };
-    }
+      if (index >= 0) {
+        if (appleSleepItem.type === SleepType.awake) {
+          sleep[index].durations = {
+            ...sleep[index].durations,
+            awake_seconds: appleSleepItem.value,
+          };
 
-    if (appleItem.type === SleepType.rem) {
-      sleepPayload.durations = {
-        rem_seconds: appleItem.value,
-      };
-    }
+          sleep[index].wakeup_frequency = sleep[index].wakeup_frequency
+            ? // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+              sleep[index].wakeup_frequency! + 1
```

Comment:
For some reason the typescript kept saying that this was undefined even though im checking that its true so i needed to add the above